Sources/Build/BuildOperation.swift

```diff
@@ -135,13 +136,17 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
         let buildSystem = try self.createBuildSystem(buildDescription: buildDescription)
         self.buildSystem = buildSystem
 
+        let buildStartTime = DispatchTime.now()
+
         buildSystemDelegate?.buildStart(configuration: self.buildParameters.configuration)
 
         // Perform the build.
         let llbuildTarget = try computeLLBuildTargetName(for: subset)
         let success = buildSystem.build(target: llbuildTarget)
 
-        buildSystemDelegate?.buildComplete(success: success)
+        let duration = buildStartTime.distance(to: .now())
```

Comment:
Half observation, half question:

Using `.now` includes time when the system sleeps. What if `.uptimeNanoseconds` is used instead, which excludes time when the system sleeps?

I don't know which is more sensible to the user. It's good to think about whether sleep time should be included though imo.

Reply:
It's a good point, though it's unlikely that the system would sleep in the middle of a build.  But that's a good suggestion and might in fact be more correct.